spec/lib/msf/core/rpc/v10/job_state_tracker_spec.rb

```diff
@@ -0,0 +1,159 @@
+require 'spec_helper'
+
+require 'msf/core/rpc/v10/job_state_tracker'
+
+RSpec.describe JobStateTracker do
+
+  context "With a 0.1 second ttl " do
+    before(:each) do
+      @job_state_tracker = JobStateTracker.new(result_ttl=0.1)
+    end
+
+    context "A job has completed" do
+      before(:each) do
+        @job_id = "super_random_job_id"
+        @generic_result = "This is a job that has finished"
+        @job_state_tracker.completed(@job_id, @generic_result)
+      end
+
+      it 'should show as finished' do
+        expect(@job_state_tracker.finished? @job_id).to eq(true)
+      end
+
+      context "we wait out the ttl" do
+        before(:each) do
+          sleep 0.1
+        end
+
+        it 'should not show as finished' do
+          expect(@job_state_tracker.finished? @job_id).to eq(false)
```

Comment:
iirc predicate methods exposed on objects can be called directly using rspec:

```ruby
expect(@job_state_tracker).to be_finished
```

https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/predicate-matchers

Which should hopefully improve readability 🎉

Reply:
ah nice, didn't know about that, looks like a good change